src/qamqpchannelhash.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2012-2014 Alexey Shcherbakov
+ * Copyright (C) 2014-2015 Matt Broadstone
+ * Contact: https://github.com/mbroadst/qamqp
+ *
+ * This file is part of the QAMQP Library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+#include "qamqpqueue.h"
+#include "qamqpexchange.h"
+#include "qamqpchannelhash_p.h"
+
+/*!
+* Retrieve a pointer to the named channel.
+*
+* A NULL string is assumed to be equivalent to "" for the purpose
+* of retrieving the nameless (default) exchange.
+*
+* \param[in]   name    The name of the channel to retrieve.
+* \retval      NULL    Channel does not exist.
+*/
+QAmqpChannel* QAmqpChannelHash::get(const QString& name) const
+{
+    if (name.isNull())
```

Comment:
``|| name.isEmpty()``, that translates to "is the QString default constructed (QString()), OR is it an empty string ''"